spec/acceptance/agent_spec.rb

```diff
@@ -87,4 +98,127 @@ class { 'zabbix::agent':
+        it { is_expected.to be_enabled }
+      end
+
+      describe file('/etc/zabbix/zabbix_agentd2.conf') do
+        its(:content) { is_expected.not_to match %r{ListenIP=} }
+      end
+    end
+
+    context 'With ListenIP set to an IP-Address' do
+      it 'works idempotently with no errors' do
+        pp = <<-EOS
+          class { 'zabbix::agent':
+            agent_configfile_path => '/etc/zabbix/zabbix_agent2.conf',
+            include_dir           => '/etc/zabbix/zabbix_agent2.d',
+            include_dir_purge     => false,
+            zabbix_package_agent  => 'zabbix-agent2',
+            servicename           => 'zabbix-agent2',
+            manage_startup_script => false,
+            server                => '192.168.20.11',
+            zabbix_package_state  => 'latest',
+            listenip              => '127.0.0.1',
+            zabbix_version        => '#{version}',
+          }
+        EOS
+
+        prepare_host
+
+        apply_manifest(pp, catch_failures: true)
+        apply_manifest(pp, catch_changes: true)
+      end
+
+      describe file('/etc/zabbix/zabbix_agent2.conf') do
+        its(:content) { is_expected.to match %r{ListenIP=127.0.0.1} }
+      end
+    end
+
+    context 'With ListenIP set to lo' do
+      it 'works idempotently with no errors' do
+        pp = <<-EOS
+          class { 'zabbix::agent':
+            agent_configfile_path => '/etc/zabbix/zabbix_agent2.conf',
+            include_dir           => '/etc/zabbix/zabbix_agent2.d',
+            include_dir_purge     => false,
+            zabbix_package_agent  => 'zabbix-agent2',
+            servicename           => 'zabbix-agent2',
+            manage_startup_script => false,
+            server                => '192.168.20.11',
+            zabbix_package_state  => 'latest',
+            listenip              => 'lo',
+            zabbix_version        => '#{version}',
+          }
+        EOS
+
+        prepare_host
+
+        apply_manifest(pp, catch_failures: true)
+        apply_manifest(pp, catch_changes: true)
+      end
+
+      context 'With ListenIP set to an IP-Address' do
+        it 'works idempotently with no errors' do
+          pp = <<-EOS
+            class { 'zabbix::agent':
+              agent_configfile_path => '/etc/zabbix/zabbix_agent2.conf',
+              include_dir           => '/etc/zabbix/zabbix_agent2.d',
+              include_dir_purge     => false,
+              zabbix_package_agent  => 'zabbix-agent2',
+              servicename           => 'zabbix-agent2',
+              manage_startup_script => false,
+              server                => '192.168.20.11',
+              zabbix_package_state  => 'latest',
+              listenip              => '127.0.0.1',
+              zabbix_version        => '#{version}',
+            }
+          EOS
+
+          prepare_host
+
+          apply_manifest(pp, catch_failures: true)
+          apply_manifest(pp, catch_changes: true)
```

Comment:
Can you please use our shared example please? https://github.com/voxpupuli/voxpupuli-acceptance?tab=readme-ov-file#an-idempotent-resource

Reply:
I'm not able to run function prepare_host in it_behaves_like like I do in the standard it 
I've try with "prepare_host" before the it_behaves_like, with a "before do" statement but it apply the code every time

continue trying, but if you have the solution please share it